packages/chopsticks/src/blockchain/block-builder.ts

```diff
@@ -287,6 +289,8 @@ export const buildBlock = async (
     storageDiff,
   })
 
+  await head.chain.saveBlockToDB(finalBlock)
```

Comment:
this shouldn't be here. create a method on Blockchain.class `onNewBlock` which deals with new blocks and replace txpool `await this.#chain.setHead(newBlock)` with `await this.#chain.onNewBlock(newBlock)`

Reply:
```diff
+ async onNewBlock(block: Block): Promise<void> {
+   await this.setHead(block);
+   await this.saveBlockToDB(block);
+ }
```

do you mean adding a `onNewBlock` method like this?